File: 2025-Winter/Vibration/HW 5_2025 data files/tutDec10_load_files.m
clearvars, clc

M = readmatrix("mass_matrixMATLAB.xlsx");
K = readmatrix("stiffness_matrixMATLAB.xlsx");
C = readmatrix("damping_matrixMATLAB.xlsx");

format long
disp('Upper left sub-half-matrix of K')
K(1:4,1:4)
disp('Lower right sub-half-matrix of K')
K(5:8,5:8)
disp('Upper right sub-half-matrix of K')
K(1:4,5:8)

shouldbezero8times8 = abs(K - transpose(K));
disp('shouldbezero8times8:')
for ii = 1:8
    fprintf(' %12.10f',shouldbezero8times8(ii,:))
    fprintf('\n')
end
fprintf('\n')

disp('Upper left sub-half-matrix of M')
M(1:4,1:4)
disp('Lower right sub-half-matrix of M')
M(5:8,5:8)
disp('Upper right sub-half-matrix of M')
M(1:4,5:8)

shouldbezero8times8 = abs(M - transpose(M));
disp('shouldbezero8times8:')
for ii = 1:8
    fprintf(' %12.10f',shouldbezero8times8(ii,:))
    fprintf('\n')
end
fprintf('\n')

disp('Upper left sub-half-matrix of C')
C(1:4,1:4)
disp('Lower right sub-half-matrix of C')
C(5:8,5:8)
disp('Upper right sub-half-matrix of C')
C(1:4,5:8)

shouldbezero8times8 = abs(C - transpose(C));
disp('shouldbezero8times8:')
for ii = 1:8
    fprintf(' %12.10f',shouldbezero8times8(ii,:))
    fprintf('\n')
end
fprintf('\n')

File: 2025-Winter/Vibration/HW 5_2025 data files/tutDec10_load_files.py
#!/usr/bin/env python
# coding: utf-8

# In[1]:


from pandas import read_excel
from printarray import printarray


# In[2]:


KDF = read_excel('stiffness_matrix.xlsx')
MDF = read_excel('mass_matrix.xlsx')
CDF = read_excel('damping_matrix.xlsx')


# In[3]:


K = KDF.values
M = MDF.values
C = CDF.values


# In[4]:


print('Upper left sub-half-matrix of K')
for ii in range(4):
    for jj in range(4):print('{:.15f}  '.format(K[ii,jj]),end="")

    print('')

print('\nLower right sub-half-matrix of K')
for ii in range(4,8):
    for jj in range(4,8):print('{:.15f}  '.format(K[ii,jj]),end="")

    print('')

print('\nUpper right sub-half-matrix of K')
for ii in range(4):
    for jj in range(4,8):print('{:.15f}  '.format(K[ii,jj]),end="")

    print('')


# In[5]:


shouldbezero8times8 = abs(K - K.T)
print('shouldbezero8times8:')
for ii in range(8):
    printarray(shouldbezero8times8[ii,:],'.10f')


# In[6]:


print('\nUpper left sub-half-matrix of M')
for ii in range(4):
    for jj in range(4):print('{:.15f}  '.format(M[ii,jj]),end="")

    print('')

print('\nLower right sub-half-matrix of M')
for ii in range(4,8):
    for jj in range(4,8):print('{:.15f}  '.format(M[ii,jj]),end="")

    print('')


print('\nUpper right sub-half-matrix of M')
for ii in range(4):
    for jj in range(4,8):print('{:.15f}  '.format(M[ii,jj]),end="")

    print('')


# In[7]:


shouldbezero8times8 = abs(M - M.T)
print('shouldbezero8times8:')
for ii in range(8):
    printarray(shouldbezero8times8[ii,:],'.10f')


# In[8]:


print('\nUpper left sub-half-matrix of C')
for ii in range(4):
    for jj in range(4):print('{:.15f}  '.format(C[ii,jj]),end="")

    print('')

print('\nLower right sub-half-matrix of C')
for ii in range(4,8):
    for jj in range(4,8):print('{:.15f}  '.format(C[ii,jj]),end="")

    print('')


print('\nUpper right sub-half-matrix of C')
for ii in range(4):
    for jj in range(4,8):print('{:.15f}  '.format(C[ii,jj]),end="")

    print('')


# In[9]:


shouldbezero8times8 = abs(C - C.T)
print('shouldbezero8times8:')
for ii in range(8):
    printarray(shouldbezero8times8[ii,:],'.10f')

